### PR TITLE
0110 mapping coalesce

### PR DESCRIPTION
**Pull request by ramiromagno**
Thank you for your Pull Request! We have developed this task checklist from the
[Development Process
Guide](https://pharmaverse.github.io/sdtm.oak/CONTRIBUTING.html)
to help with the final steps of the process. Completing the below tasks helps to
ensure our reviewers can maximize their time on your code as well as making sure
the oak codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task
or check off that it is not relevant to your Pull Request. This checklist is
part of the Github Action workflows and the Pull Request will not be merged into
the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull z
Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the 
[tidyverse style guide](https://style.tidyverse.org/). Run 
`styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should
consider realistic data scenarios and edge cases, e.g. empty datasets, errors,
boundary cases etc. - See 
[Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you 
fully follow the 
[deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords
and families. Refer to the 
[categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the
`NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it
has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build oak site `pkgdown::build_site()` and check that all affected
examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/assign.R [hunk 1]
```diff
@@ -61,8 +61,11 @@ sdtm_assign <- function(tgt_dat = NULL,
   # Recode the raw variable following terminology.
   tgt_val <- ct_map(join_dat[[raw_var]], ct_spec = ct_spec, ct_clst = ct_clst)
 
+  # Current target values
+  cur_tgt_val <- join_dat[[tgt_var]] %||% as.vector(NA, mode = typeof(tgt_val))
+
   join_dat |>
-    mutate("{tgt_var}" := tgt_val) |> # nolint object_name_linter()
+    mutate("{tgt_var}" := dplyr::coalesce(cur_tgt_val, tgt_val)) |> # nolint object_name_linter()
```
Comment (ShiyuC) [inline R/assign.R:68]:
```suggestion
    dplyr::mutate({{tgt_var}} := dplyr::coalesce(cur_tgt_val, tgt_val)) |> # nolint object_name_linter()
```
@ramiromagno I tested the branch using the mock data I prepared for the PharmaSUG workshop, and it seems that the coalesced values are not correctly assigned to the target variable name tgt_var. I went in and debugged this function and realized it would work with {{}} symbol. Please consider the code suggestion above. Thank you!

**Comment by ramiromagno** [comment]
Hi @ShiyuC ,

thanks for looking into this.

Could you make a reprex of what is not working exactly? I'm asking because Ramm's test example was passing the unit test just fine...

https://github.com/pharmaverse/sdtm.oak/pull/114/files#diff-63d348a1869108982a9b12f5dd68b346b6d159bbcd388e88122bd337d326b8a3

**Comment by ramiromagno** [comment]
Could you run your code with `reprex::reprex()`? And perhaps make it more isolated on the specific mapping if believe not to be working? The `study_ct` is not defined in the code above.

File: R/cnd_df.R [hunk 2]
```diff
@@ -160,7 +160,7 @@ rm_cnd_df <- function(dat) {
     attr(dat, "cnd") <- NULL
     attr(dat, "cnd_sum") <- NULL
   }
-  return(dat)
+  dat
```
Comment (rammprasad) [inline R/cnd_df.R:163]: @ramiromagno - Can I ask why you decided to change the `return(dat)` to `dat`

File: tests/testthat/test-coalesce-behavior.R [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  assign_ct(
+    raw_dat = cm_raw,
+    raw_var = "IT.CMDOSU",
+    tgt_var = "CMDOSU",
+    ct_spec = study_ct,
+    ct_clst = "C71620",
+    id_vars = oak_id_vars()
+  )
+
+cm_expected <-
```
Comment (ShiyuC) [inline tests/testthat/test-coalesce-behavior.R:154]: Would it be possible to align this tibble for better readability?

File: R/cnd_df.R (hunk 2 above)
Comment (ShiyuC) [reply R/cnd_df.R:163]: Is it because lint complains about explicit return?

**Review by ShiyuC — APPROVED** [review]
Looks good to me Ramiro. Just left one minor suggestion for your consideration

File: tests/testthat/test-coalesce-behavior.R (hunk 3 above)
Comment (ramiromagno) [reply tests/testthat/test-coalesce-behavior.R:154]: I will align, no problem. Not sure it's going to improve readability :)

File: R/cnd_df.R (hunk 2 above)
Comment (ramiromagno) [reply R/cnd_df.R:163]: indeed!

File: tests/testthat/test-coalesce-behavior.R (hunk 3 above)
Comment (ramiromagno) [reply tests/testthat/test-coalesce-behavior.R:154]: done. Thanks for the feedback!
